pkg/maintainer/wallet/wallet_test.go

```diff
@@ -0,0 +1,72 @@
+package wallet
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/keep-network/keep-core/pkg/tbtc"
+)
+
+func TestRunIfWalletUnlocked_WhenLocked(t *testing.T) {
+	localChain := newLocalChain()
+
+	walletPublicKeyHash := [20]byte{1}
+
+	lockExpiration := time.Now().Add(500 * time.Millisecond)
+
+	localChain.setWalletLock(
+		walletPublicKeyHash,
+		lockExpiration,
+		tbtc.Heartbeat,
+	)
+
+	runFunc := func() error {
+		return fmt.Errorf("boom, you should not run me")
+	}
+
+	wm := &walletMaintainer{
+		config:   Config{},
+		chain:    localChain,
+		btcChain: newLocalBitcoinChain(),
```

Comment:
We could also do `btcChain: nil, // not relevant in this scenario` and avoid the entire `pkg/maintainer/wallet/bitcoin_chain_test.go` construction. I don't think we'll need to inspect Bitcoin chain for redemption requests either.

Reply:
ad2dafe1904a2c4ddafd60d062d3e27b06530275